modules/fs.js

```diff
@@ -158,8 +153,28 @@ function open(path, options) {
     if (!read && !write && !append && !update) {
         read = true;
     }
+
+    // configure the NIO options
+    var nioOptions = new java.util.ArrayList();
+    if (append === true) {
+        nioOptions.add(java.nio.file.StandardOpenOption.APPEND);
+    }
+    if (read === true) {
+        nioOptions.add(java.nio.file.StandardOpenOption.READ);
+    }
+    if (write === true) {
+        nioOptions.add(java.nio.file.StandardOpenOption.WRITE);
+        nioOptions.add(java.nio.file.StandardOpenOption.CREATE);
+        nioOptions.add(java.nio.file.StandardOpenOption.TRUNCATE_EXISTING);
+    }
+
+    // needed to generate a valid varargs argument for the Java NIO API
+    var javaVarArgs = nioOptions.toArray(
+        java.lang.reflect.Array.newInstance(java.nio.file.OpenOption, nioOptions.size())
+    );
+
```

Comment:
Java varargs methods accept a JS array as argument, so you could simplify to `[StandardOptions.WRITE, StandardOptions.CREATE, ...]` without going through the hassle of using `nioOptions.toArray()` or instantiating an `ArrayList`.

Reg. readability you could use `StandardOpenOption` as part of a destructuring assingment, eg.

    var {Files, Paths, StandardOpenOption} = java.nio.file;

same with

    var {File, FileInputStream, FileOutputStream} = java.io;